File: src/endpoints/file/delete_item.py
from src import output
from src.libs.db import DB, Table
from src.modules.verify import checking_token
from . import __types__ as types, __libs__ as file_libs

async def delete_item(params: dict = {}, req = None, _db: DB = None, _token = None):

    # set values
    result = None
    db = _db if _db else DB().connect()

    try:
        # set params
        params = types.DeleteItem(**params)

        # checking token
        token = checking_token(req, db) if not _token else _token

        # set where
        where = [ f'srl = {params.srl}' ]

        # get item
        item = db.get_item(
            table_name=Table.FILE.value,
            fields=[ 'srl', 'code', 'path' ],
            where=where,
        )
        if not item: raise Exception('Item not found.', 204)

        # delete item
        db.delete_item(
            table_name=Table.FILE.value,
            where=where,
        )

        # delete file
        file_libs.delete_file(item['path'])

        # delete cache files
        file_libs.delete_cache_files(item.get('code'))

        # set result
        result = output.success({
            'message': 'Complete delete file.',
        }, _req=req)
    except Exception as e:
        result = output.exc(e, _req=req)
    finally:
        if not _db and db: db.disconnect()
        return result

File: src/routes/file/delete-item.ts
import DB, { db } from '@/classes/DB'
import ServiceError from '@/classes/ServiceError'
import { deleteFile } from '@/libs/file'
import * as helper from './__helper'

export default async function deleteItem(srl: number)
{
  let _transaction = false
  try
  {
    // set assets
    let _where = `srl = ${srl}`

    // begin transaction
    _transaction = db.transaction('begin')

    // get item
    const item = db.getData({
      table: DB.TABLE.FILE,
      field: 'srl,code,path',
      where: _where,
    })
    if (!item.data)
    {
      throw new ServiceError('No data', { status: 204 })
    }

    // delete data
    db.deleteData({
      table: DB.TABLE.FILE,
      where: _where,
    })

    // delete file
    await deleteFile(item.data.path)

    // delete cache files
    await helper.deleteCache(item.data.code)

    // commit transaction
    _transaction = db.transaction('commit')
  }
  catch (_e: any)
  {
    // collback transaction
    db.transaction('rollback', _transaction)

    throw new ServiceError('Failed to delete File.', {
      status: _e.status,
      text: _e.message,
      cause: _e,
    })
  }
}
